lib/src/facts/openbsd/virtualization_resolver.cc

```diff
@@ -22,4 +22,10 @@ namespace facter { namespace facts { namespace openbsd {
         return {};
     }
 
+    string virtualization_resolver::get_cloud_provider(collection& facts)
+    {
+        // TODO Implement this
+        return "";
```

Comment:
In theory the pattern is, only fill what elements you can of the data struct, then check each to make sure it exists before adding a corresponding value to the collection. So really, what this returns could be arbitrary, depending on how it's handled in the collect_data method.